herddb-core/src/main/java/herddb/sql/CalcitePlanner.java

```diff
@@ -784,8 +784,35 @@ private PlannerOp planEnumerableTableScan(EnumerableTableScan scan, RelDataType
                 (TableImpl) scan.getTable().unwrap(org.apache.calcite.schema.Table.class
                 );
         Table table = tableImpl.tableManager.getTable();
-        ScanStatement scanStatement = new ScanStatement(tableSpace, table, null);
-        return new TableScanOp(scanStatement);
+        Column[] columns = table.getColumns();
+        int numColumns = columns.length;
+        boolean usingAliases = false;
+        for (int i = 0; i < numColumns; i++) {
+            String alias = rowType.getFieldNames().get(i).toLowerCase();
+            String colName = columns[i].name;
+            if (!alias.equals(colName)) {
+                usingAliases = true;
+                break;
+            }
+        }
+        if (usingAliases) {
+            String[] fieldNames = new String[numColumns];
+            int[] projections = new int[numColumns];
+            for (int i = 0; i < numColumns; i++) {
+                String alias = rowType.getFieldNames().get(i).toLowerCase();
+                fieldNames[i] = alias;
+                projections[i] = i;
+            }
+            Projection zeroCopy = new ProjectOp.ZeroCopyProjection(
```

Comment:
we could create a new AliasedIdentityProjection but it will be used only in this very specific case, I am not sure it is worth.
ZeroCopyProjection is widely used and tested and it introduces only a minimal cost.